pkg/controller/actions/deploy/action_deploy.go

```diff
@@ -221,10 +232,20 @@ func (a *Action) deploy(
 		return false, fmt.Errorf("failed to lookup object %s/%s: %w", obj.GetNamespace(), obj.GetName(), lookupErr)
 	}
 
-	// the user has explicitly marked the current object as not owned by the operator, so
-	// skip any further processing
-	if current != nil && resources.GetAnnotation(current, annotations.ManagedByODHOperator) == "false" {
-		return false, nil
+	if current != nil {
+		// Remove the DSC owner reference if set, This is required during the
+		// transition from the old to the new operator.
+		if err := removeOwnerReferences(ctx, rr.Client, current, func(or metav1.OwnerReference) bool {
+			return or.APIVersion == gvk.DataScienceCluster.GroupVersion().String() && or.Kind == gvk.DataScienceCluster.Kind
```

Comment:
For deploy action should we also check for DSCI kind?

Reply:
done